lua/lspconfig/util.lua

```diff
@@ -405,4 +396,9 @@ function M.path.exists(filename)
   return stat and stat.type or false
 end
 
+--- @deprecated use `vim.fs.find('.hg', { path = startpath, upward = true })` instead
+function M.find_mercurial_ancestor(startpath)
+  return vim.fs.find('.hg', { path = startpath, upward = true })
```

Comment:
A limit should be set here. And it is possible that multiple matching directories will be returned. The first one found may not be the correct root directory

Reply:
Wait what? How does one identify a "real" root directory? I assumed it meant the first parent directory containing ".hg"?

Reply:
Oh.. I misread this. I was confused and thought it was node search. Sorry I haven't slept for 35 hours so this is a bit confusing. nvm